sym/provider/flow_resource.go

```diff
@@ -276,15 +277,46 @@ func readFlow(_ context.Context, data *schema.ResourceData, meta interface{}) di
 	// Base64 -> Text
 	diags = utils.DiagsCheckError(diags, data.Set("implementation", utils.ParseRemoteImpl(flow.Implementation)), "Unable to read Flow implementation")
 
-	// Terraform block is called "prompt_field", so that's what Terraform expects. The Sym API returns
-	// "prompt_fields", so change the key before giving it to Terraform.
-	if promptFields, found := flow.Params["prompt_fields"]; found {
-		flow.Params["prompt_field"] = promptFields
-		delete(flow.Params, "prompt_fields")
+	// The API may add new parameters that we (the provider) don't know about yet, so rebuild the map and include
+	// only the parameters we do know about. Otherwise, the data.Set below will fail to set the Flow's state.
+	knownParams := map[string]interface{}{}
+	paramsSchema := flowParamsResource().Schema
+	promptFieldSchema := promptFieldResource().Schema
+
+	for paramKey, paramValue := range flow.Params {
+		if paramKey == "prompt_fields" {
+			// Do the same check for known parameters within each prompt field
+			var knownPromptFields []map[string]interface{}
+
+			// Iterate over each prompt field from the API response
+			for i := range paramValue.([]interface{}) {
+				promptField := paramValue.([]interface{})[i].(map[string]interface{})
+				knownPromptField := map[string]interface{}{}
```

Comment:
what the hell is this

`map[string]interface{}{}`

is this real?

Reply:
hahahaha. this is actually fairly standard in go land.

`map[string]interface{}` is a type that means "map of string to any", basically.
and then you initialize it as an actual object with `{}` so `map[string]interface{}{}` is initializing a map of string to any